Add some backward compatible code for IsWindows and IsLinux

harrisonmeister opened the pull request:
We had an [internal report](https://octopusdeploy.slack.com/archives/CC44LTNF5/p1646818680709189) about cloning a tentacle instance in the incorrect folder. Presumably, this was on running on Windows PowerShell.

Not a big fan of even having to add this, but it does seem to work.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OctopusDeployLabs/SpaceCloner?pullRequest=35) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OctopusDeployLabs/SpaceCloner?pullRequest=35) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OctopusDeployLabs/SpaceCloner?pullRequest=35) it.</sub>

BobJWalker reviewed (changes requested):
This looks to be a bit overkill.  IsWindows and IsLinux was added in PowerShell Core 6.  PowerShell 5.1 and lower wouldn't have those variables.  But those versions of PS only run on Windows.

If those variables are not set then it is safe to assume they are running on Windows.

harrisonmeister commented:
> This looks to be a bit overkill. IsWindows and IsLinux was added in PowerShell Core 6. PowerShell 5.1 and lower wouldn't have those variables. But those versions of PS only run on Windows.
> 
> If those variables are not set then it is safe to assume they are running on Windows.

@BobJWalker - I've removed the code I added , and just switched around the one place where testing for `$IsWindows` was failing to default to Windows.

BobJWalker reviewed (approved):
LGTM
